### PR TITLE
[IMP] *: improve demo and command barcodes

### PR DESCRIPTION
**Pull request by mohammadhossam**
*: barcodes, mrp{_subcontracting}, repair, stock, web

Revamp the barcodes of commands and demo data to be shorter and with no special characters. So that they are more readable and easier to remember. Also all demo products barcodes are changed to be EAN13 compliant, so the missing check digit is added.

ENT PR: odoo/enterprise#59242

Task-3774096

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/tests/legacy/mobile/views/fields/many2one_barcode_field_tests.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ QUnit.module("Fields", (hooks) => {
                         {
                             id: 111,
                             name: "product_cable_management_box",
-                            barcode: "601647855631",
+                            barcode: "6016478556318",
```
Comment (svs-odoo) [inline addons/web/static/tests/legacy/mobile/views/fields/many2one_barcode_field_tests.js:44]: Why this change ?

Comment (svs-odoo) [reply addons/web/static/tests/legacy/mobile/views/fields/many2one_barcode_field_tests.js:44]: Alright but this is not a demo data, it's a data used by QUnit tests only, I don't think it makes a difference to update them

Comment (mohammadhossam) [reply addons/web/static/tests/legacy/mobile/views/fields/many2one_barcode_field_tests.js:44]: Alright, I thought they are affected by demo data since they have the same exact barcode

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo r+
